### PR TITLE
Plans: Show ability to set custom domain as primary under plan benefits

### PR DESCRIPTION
**Pull request by eltongo**
#### Changes proposed in this Pull Request

This PR emphasizes the ability to set a domain as primary under plan benefits after upgrading, as part of this test: p99Zz8-10j-p2

Screenshot 2020-03-04 at 4 02 39 PM

#### Testing instructions

* Purchase a plan for a site
* Make sure the card is not shown before purchasing a domain
* Claim your free domain
* Click Plan on your sidebar
* Verify that the card is shown like in the screenshot above after you register the domain under plan features, and that the button sends you to the right place
* Make sure that the card is not shown if a custom domain is set as primary
* Verify that nothing gets broken

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx [hunk 1]
```diff
@@ -73,6 +100,19 @@ CustomDomainPurchaseDetail.propTypes = {
...
+export default connect( state => {
+	const siteId = getSelectedSiteId( state );
+	const siteDomains = getDomainsBySiteId( state, siteId );
+	const registeredDomains = getRegisteredDomains( siteDomains );
+
+	return {
+		hasNonPrimaryDomainsFlag: getCurrentUser( state )
+			? currentUserHasFlag( state, NON_PRIMARY_DOMAINS_TO_FREE_USERS )
+			: false,
+		registeredDomain: head( registeredDomains ),
```
Comment (daledupreez) [inline client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx]: This assumption that the first item in this list feels like it's really brittle, as if I've bought more than one domain, the first one in the list could easily be the wrong one. I think we either need to fix the language here to drop the "new" in "Make your **new** domain your primary address", or update the logic to actually find the newly purchased domain.

@fditrapani, do you have a preference on which way this ought to go?

Comment (eltongo) [reply client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx]: Honestly I'd rather have the "new" part dropped, though "new" would apply to any domain the user would purchase during the month of this test.

**Comment by daledupreez** [comment]
@eltongo, I did the following on `calypso.live` and did _not_ see the expected card:

1. I started from a free site using my account, which went through fine.
2. I bought a domain, and this didn't bundle a plan.
3. I then bought a Personal plan, but did *not* see the card during checkout. (Issue 1)
4. I then bought a domain using the domain credit on my plan.
5. When I arrived in checkout, I saw the `Your primary domain` card that indicated the domain I had bought in step 2 was currently the primary domain. (Issue 2) I assume this happened in the background.

Note that I was not running in a sandbox, so I am not sure what I ought to have seen here.

**Comment by eltongo** [comment]
@daledupreez my bad for not making this clear enough. This is supposed to appear under plan features/benefits when clicking "Plan" on your sidebar, not in the checkout.

**Pull request merged**